### DISCUSSION TITLE
🐞 go opentelemetry build errors in v0.13.0

### DISCUSSION THREAD

**Discussion by DarkHeros09** [dagger/dagger#12208] [Help]
### What is the issue?

Hi

I'm getting this error during test
```
Error: ../../../go/pkg/mod/dagger.io/dagger@v0.13.0/telemetry/exporters.go:92:23: cannot use log (variable of type "go.opentelemetry.io/otel/sdk/log".Record) as *"go.opentelemetry.io/otel/sdk/log".Record value in argument to e.OnEmit
```

### Dagger version

dagger v0.13.0

**Comment by devexter** [comment]
I am seeing the same error after upgrading my go dagger module to 0.13.0

**Comment by jedevc** [comment]
cc @vito 

I think this happens because we pin the OTEL logging, but *sometimes* the base OTEL stack can upgraded (e.g. if using a parent `go.mod`, or dependabot decides to do it, etc) because we don't pin it.

Should we just pin all of OTEL for modules? Sucks, but given how unstable the upstream APIs are, I'm wondering if we should, as suggested here: https://github.com/dagger/dagger/pull/8234#discussion_r1732585400

**Comment by vito** [comment]
@jedevc We already pin this dependency for modules, but the issue still remains for non-module use cases, which seems to be the situation here considering the error path (`../../../go/pkg/mod/dagger.io/dagger@v0.13.0/telemetry/exporters.go`).

For anyone hitting this with non-modules, you could copy-paste these `replace` rules - ugly solution, but it would prevent the dependency from drifting forward into brokenness.

https://github.com/dagger/dagger/blob/06839862e3437c2f16d95c3621e0d7b1f7e50ab9/sdk/go/go.mod#L55-L60

Maybe we could just vendor these dependencies until they stabilize? It's been a while since I've done that, don't remember how it interacts with Go modules and packages. 🤔 

Lastly, we should just bump our OTel dependencies.

@devexter If you're seeing it happen for a Dagger module, try running `dagger develop` so that it adds those `replace` rules locally.
